xcom2-launcher/xcom2-launcher/Classes/Steam/SteamManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Steamworks;
+
+namespace XCOM2Launcher.Steam
+{
+    public static class SteamManager
+    {
+        private static bool _initialized;
+        private static readonly object _initlock = new object();
+        private static readonly System.Timers.Timer _timer = new System.Timers.Timer(50);
+
+        static SteamManager()
+        {
+            _timer.Elapsed += (sender, args) =>
+            {
+                SteamAPI.RunCallbacks();
+            };
+        }
+        
+        public static bool IsSteamRunning()
+        {
+            return SteamAPI.IsSteamRunning();
+        }
+        
+        public static bool EnsureInitialized()
+        {
+            if (_initialized) return true;
+            if (!SteamAPI.IsSteamRunning()) return false;
+
+            lock (_initlock)
+            {
+                if (_initialized) return true;
+
+                SteamAPI.Init();
+                _timer.Start();
+                _initialized = true;
+            }
+
+            return true;
+        }
+
+        public static void Shutdown()
+        {
+            if (!_initialized) return;
```

Comment:
_initialized should only be accessed from inside synchronized block for consistency. This line could probably just be removed.

Reply:
same: double-checked lock pattern.